flow/designs/asap7/mock-array-big/macro-placement.tcl

```diff
@@ -0,0 +1,18 @@
+# fish out values from single source of truth: config.py
+set vals [regexp -all -inline {\S+} [exec sh -c {cd designs/asap7/mock-array-big && python3 -c "import config;print(f'{config.rows} {config.cols} {config.x_pitch} {config.offset_x} {config.y_pitch} {config.offset_y}')"}]]
```

Comment:
where is "import config" satisfied from?

Reply:
nvm I see it now.  Why not move that to tcl?

Reply:
I would rather program in Python.

I'm now seeing that the https://en.wikipedia.org/wiki/Inversion_of_control used by ORFS is making it hard to do a number of things. 

Example:

- two macros of different types need to share pin layout, how to do this? First one macro needs to be built, then in the IO pin placement of the second macro, information from the first macro needs to be used.
- it is tricky to communicate the floorplan to small scriptlets that are being invoked here and there(like die area size)

Reply:
You could add a MACRO_PLACEMENT_PY option and do the manual placement from python.  The python API is still under development but for a simple use case like this one it should suffice.